tools/smithy-rs-sync/src/main.rs

```diff
@@ -66,13 +66,26 @@ fn main() {
         branch,
     } = Opt::from_args();
 
-    if let Err(e) = sync_aws_sdk_with_smithy_rs(&smithy_rs, &aws_sdk, &branch) {
-        eprintln!("Sync failed with error: {:?}", e);
-    };
+    match sync_aws_sdk_with_smithy_rs(&smithy_rs, &aws_sdk, &branch) {
+        Ok(_) => std::process::exit(0),
+        Err(e) => {
+            eprintln!("Sync failed with error: {:?}", e);
+            std::process::exit(1);
+        }
+    }
 }
```

Comment:
fwiw, I think you can just have `main` return a Result. Probably more future proof